File: Climbing Stairs.cpp
#define F(i, a, b) for(int i = a; i <= b; i++)
#define FF(i, a, b) for( i = a; i <= b; i++)
#define F3(i, a, b) for(int i = a; i <= b; i+=3)
#define FB(i, a, b) for(int i = a; i >= b; i--)
#define MAX2(a, b) a>b?a:b
#define MAX3( a, b, c ) a>b?(a>c?a:c):(b>c?b:c)
#define MIN2(a, b) a<b?a:b

#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <tuple>
#include <iterator>
#include <stack>
#include <algorithms>

class Solution {
public:
    int climbStairs(int n) {
        vector<int> ret = vector<int>( n+1, 1 );
        F(i, 2, n){
        	ret[i] = ret[i-2] + ret[i-1];
        }

        return ret[n]; 
    }
};

File: LeetCode-Java/Climbing Stairs.java
public class Solution {
	public static int climbStairs(int n) {
		int [] dp = new int [n+1];
		dp[0] = 1;

		for( int i = 0; i <= n-2; i++ ){
			dp[i+1] += dp[i];
			dp[i+2] += dp[i];
		}

		dp[n] += dp[n-1];
		return dp[n];
	}
}
